### PR TITLE
Skip item merges if stack size is above/equal to the max stack size

### LINKED ISSUE #1211: Too lazy to make a PR as I'd have to build 1.12, Free item optimization
You can reduce a lot of item lag by skipping merges if the stack size is above/equal to the max stack size

### PR DESCRIPTION
**Pull request by hugmanrique**
Closes #1211

### CODE REVIEW

File: Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+index 0b7fc327..1cc1acee 100644
+--- a/src/main/java/net/minecraft/server/EntityItem.java
++++ b/src/main/java/net/minecraft/server/EntityItem.java
+@@ -177,6 +177,14 @@ public class EntityItem extends Entity implements HopperPusher {
+     // Spigot end
+ 
+     private void x() {
++        // Paper start - avoid item merge if stack size above max stack size
++        ItemStack stack = getItemStack();
++
```
Comment (mikroskeem) [inline Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch]: You can make the diff smaller by removing empty newlines

File: Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
++        if (stack.getCount() >= stack.getMaxStackSize()) {
++            return;
++        }
```
Comment (mikroskeem) [inline Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch]: ... and by removing the brackets, as you are doing `return` anyway.

**Review by mikroskeem — COMMENTED** [review]
Don't forget to squash that commit to previous one to make git history pretty :smile:

**Review by aikar — APPROVED** [review]

**Review by mikroskeem — APPROVED** [review]

File: Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch [hunk 3]
```diff
@@ -0,0 +1,25 @@
+From 311e9503ab54d720ccf3304c144ea85fb9e4ca93 Mon Sep 17 00:00:00 2001
+From: Hugo Manrique <hugmanrique@gmail.com>
```
Comment (stephan-gh) [inline Spigot-Server-Patches/0330-Avoid-item-merge-if-stack-size-above-max-stack-size.patch:2]: ~~Note: This is not the same email you added to the MIT license file. You probably want to add this mail to to `LICENSE.md` or reconfigure Git to use the other mail address.~~

**Review by stephan-gh — CHANGES_REQUESTED** [review]

**Comment by hugmanrique** [comment]
It should be fine now! Thanks for the help, this was the first time I worked on a patch-based setup 😅

**Review by kashike — APPROVED** [review]

**Review by zachbr — APPROVED** [review]

**Comment by zachbr** [comment]
Thanks! 👍

**Pull request merged**